exp/api/v1alpha3/azuremanagedcontrolplane_webhook.go

```diff
@@ -90,11 +93,34 @@ func (r *AzureManagedControlPlane) ValidateDelete() error {
 	return nil
 }
 
+// Validate the Azure Managed ControlPlane and return an aggregate error
+func (r *AzureManagedControlPlane) Validate() error {
+	var allErrs field.ErrorList
+	err := r.validateDNSServiceIP()
+	if err != nil {
+		allErrs = append(allErrs, err)
+	}
+
+	if r.Spec.Version != "" {
+		if _, err := semver.Parse(strings.TrimPrefix(strings.TrimSpace(r.Spec.Version), "v")); err != nil {
```

Comment:
Here you should be able to use `semver.ParseTolerant` which does the Trims for you

Reply:
done, thanks for your review :)